text/0000-vec-fallible-allocation.md

```diff
@@ -0,0 +1,276 @@
+- Feature Name: `vec_fallible_allocation`
+- Start Date: 2022-20-04
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+`Vec` has many methods that may allocate memory (to expand the underlying storage, or shrink it down). Currently each of these method rely on "infallible"
+allocation, that is any failure to allocate will call the global OOM handler, which will (typically) panic. Even if the global OOM handler does not panic, the
+return type of these method don't provide a way to indicate the failure.
+
+Currently `Vec` does have a `try_reserve` method that uses "fallible" allocation: if `try_reserve` attempts to allocate, and that allocation fails, then the
+return value of `try_reserve` will indicate that there was a failure, and the `Vec` is left unchanged (i.e., in a valid, usable state). We propose adding
+more of these `try_` methods to `Vec`, specifically for any method that can allocate. However, unlike most RFCs, we are not suggesting that this proposal is
+the best among many alternatives (in fact we know that adding more `try_` methods is undesirable in the long term), instead we are suggesting this as a way
```

Comment:
What is the rationale behind the "we know"? i.e. why do we know?

Reply:
Thanks!

Yeah, the other section explains the drawbacks, but having drawbacks does not necessarily mean the methods need to be introduced with the expectation of getting removed. Maybe it would be good to explain why we want to introduce this as a temporary solution.

Reply:
I'd rather point folks at the drawbacks and motivations section than increase the summary section any further: it's already larger than the suggested size and I'd only be repeating what the other two sections say.